libcxx/include/__charconv/to_chars_integral.h

```diff
@@ -274,6 +274,7 @@ _LIBCPP_CONSTEXPR_SINCE_CXX23 _LIBCPP_HIDE_FROM_ABI int __to_chars_integral_widt
 template <typename _Tp>
 inline _LIBCPP_CONSTEXPR_SINCE_CXX23 _LIBCPP_HIDE_FROM_ABI to_chars_result
 __to_chars_integral(char* __first, char* __last, _Tp __value, int __base, false_type) {
+# ifndef _LIBCPP_CHARCONV_OPTIMIZE_SIZE
```

Comment:
I'm not thrilled with this option. Options have a cost; they need to be tested in our CI to see whether the tests are passing when a option is added.  If you want to pursue this option I like to see the size it saves and the performance it costs. The dylib also contains part of this code, that should also be #ifdef'ed. I know you use LTO to strip the binary, but the library should not depend on that.

Reply:
Most compilers have a preprocessor define when targeting `-Os`. Would you prefer that was used?

Reply:
Yes, ideally we'd do something like automatically switch to this implementation under `-Os` (we have precedent for that in a few algorithms where we use vectorization). However, it becomes impossible to break ABI if we do that, since we *must* have a compatible ABI regardless of the optimization level.

How much of a code size benefit are we talking about here? If the code size benefits are extremely large, it might make sense to think about whether this can be made the default implementation (breaking the ABI would still be an issue but we can still entertain that idea).

Reply:
+1 for using `-Os` instead of a libc++ configuration option.

Reply:
Are you able to point me towards an example of where this is done well in libc++ so I can recreate its usage here?